packages/lodestar/src/network/peers/peerManager.ts

```diff
@@ -560,7 +560,7 @@ export class PeerManager {
       if (openCnx) {
         const direction = openCnx.stat.direction;
         peersByDirection.set(direction, 1 + (peersByDirection.get(direction) ?? 0));
-        const client = getClientFromPeerStore(openCnx.remotePeer, this.libp2p.peerStore.metadataBook);
+        const client = await getClientFromPeerStore(openCnx.remotePeer, this.libp2p.peerStore.metadataBook);
```

Comment:
Shit forgot about this one too. @tuyennhv can you keep the client also in memory in the same way that this PR? https://github.com/ChainSafe/lodestar/pull/3838 there's not need to write the client implementation to disk. Do it in another PR I can review quickly once ready and unblock this one

Reply:
@tuyennhv Just opened https://github.com/ChainSafe/lodestar/pull/3841 please take a look and feel free to commit on top if necessary

Reply:
@dapplion "AgentVersion" is written by libp2p to metadataBook and we can't control that

we can duplicate this data to our PeerManager PeerData, the issue is we also need this data in req/resp log too. Right now we always query from libp2p peerstore metadataBook

Reply:
addressed in #3841